daemon/daemon.go

```diff
@@ -589,13 +589,13 @@ func NewDaemon(c *Config) (*Daemon, error) {
 	if c.IsK8sEnabled() {
 		d.k8sClient, err = k8s.CreateClient(c.K8sEndpoint, c.K8sCfgPath)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("k8s create client: %s", err)
 		}
 
 		if nodeName := os.Getenv(k8s.EnvNodeNameSpec); nodeName != "" {
 			// Try to retrieve node's cidr from k8s's configuration
 			if err := d.useK8sNodeCIDR(nodeName); err != nil {
-				return nil, err
+				return nil, fmt.Errorf("k8s use node CIDR: %s", err)
```

Comment:
If we use `log.Fatalf` we can't unit test that function in #1039 . I've changed this error to `return nil, fmt.Errorf("unable to retrieve node CIDR: %s", err)`